foodsaving/subscriptions/consumers.py

```diff
@@ -1,31 +1,64 @@
+from base64 import b64decode
+from urllib.parse import unquote
+
 from channels.generic.websockets import JsonWebsocketConsumer
 from django.utils import timezone
+from rest_framework.authentication import TokenAuthentication
 
 from foodsaving.subscriptions.models import ChannelSubscription
 
+token_auth = TokenAuthentication()
+
+
+def check_for_auth_token_header(message):
+    prefix = 'karrot.token.value.'
+    for header, value in message['headers']:
+        if header == b'sec-websocket-protocol':
+            protocols = [x.strip() for x in unquote(value.decode('ascii')).split(",")]
+            for protocol in protocols:
+                if protocol.startswith(prefix):
+                    value = protocol[len(prefix):]
+                    if len(value) % 4:
```

Comment:
Does this happen in normal operation or is it to protect against attacks? In any way, would be good to cover the other case (multiple of 4) in a test case as well.

Reply:
Normal case, e.g. `foo` base64's to `Zm9v` (len=4), and `foo1` to `Zm9vMQ==` (len=8). The other case is standard base64, so less interesting, but I'll add a test anyhow.

Reply:
Test added!